### ISSUE TITLE
Spotifylistenalong

### ISSUE THREAD

**Issue by Willthegoatz** [ordinall/BetterDiscord-Stuff#24]
Spotifylistenalong dont work on discord i kept restarting my discord uninstalling it and installing it it doesnt work for both

**Comment by Florian-Mahlberg** [comment]
Spotifylistenalong got patched with the last discord update it seems :c

**Comment by ilorde-exe** [comment]
SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST SPOTIFY WILL NEVER GET MY MONEY I HATE THE ANTICHRIST

**Comment by ilorde-exe** [comment]
I hope you guys will fix it soon love your work!!

**Comment by Florian-Mahlberg** [comment]
I went a bit through the debugging console in discord and it seems like discord is changing the Spotify variable back to "false", when you press the "Listen along" button. Maybe I'll try to find a fix as well and submit a pr in a couple of days.

**Comment by ordinall** [comment]
I am aware of the issue. It happened because discord pushed a big update changing literally every module which essentially broke every BD plugin. I'll fix it soon, have exams right now

**Comment by Florian-Mahlberg** [comment]
Good luck with your exams ^^

**Comment by ChapperNi** [comment]
good luck with exams king

**Comment by ilorde-exe** [comment]
Focus on your exam bro...we can wait

**Comment by ordinall** [comment]
Thank you all for wishing me luck on the exams, they went great. About the plugins, I looked into it and it looks like we will have to wait till BD 1.8.0 is released. This and my other plugins rely heavily on patching modules, but patching is very broken, and near impossible on some modules. BD devs did come up with a partial way that will be included in 1.8.0. Also, when digging in discord's code, the client-side Spotify premium check logic used by discord had some changes in the update too, so that will also require me some time to figure out.

**Comment by Florian-Mahlberg** [comment]
Yeah, i figured. I think the BD devs will need a bit more time to release 1.8.0...  we can only sit here and wait i guess :/

**Comment by M2Dawn** [comment]
gl with ur exams ^^

**Comment by LemonyOwO** [comment]
1.8.0 has been released. Hope you have luck with exams!

**Comment by d0gkiller87** [comment]
Here's the working update. Replace the `onStart()` function in `SpotifyListenAlong.plugin.js` with the following snippet:
```js
onStart() {
	const getActiveSocketAndDevice = WebpackModules.getByProps( 'getActiveSocketAndDevice' );	
	if ( getActiveSocketAndDevice?.getActiveSocketAndDevice ) {
		BdApi.Patcher.after(
			'', getActiveSocketAndDevice, 'getActiveSocketAndDevice',
			( that, args, ret ) => {
				if ( ret?.socket ) ret.socket.isPremium = true;
				return ret;
			}
		);
	}
}
```
 Best wishes on your exams @ordinall :)

**Comment by Florian-Mahlberg** [comment]
@d0gkiller87 maybe you can create a pr and change the version number so everyone get's the update or do you want me to do it?

**Comment by ordinall** [comment]
Thank you @d0gkiller87 for the snippet, I did manage to find this module but patching had some issues in 1.7.0 and I haven't had time to look into it after 1.8.0 release, last time I checked the module which returned had mangled id getters for actual function. Maybe it changed now as you have directly retrieved it and patching it. I will have a look again now after the update. In the meantime feel free to send me a pull request with your changes.

**Comment by d0gkiller87** [comment]
> Thank you @d0gkiller87 for the snippet, I did manage to find this module but patching had some issues in 1.7.0 and I haven't had time to look into it after 1.8.0 release, last time I checked the module which returned had mangled id getters for actual function. Maybe it changed now as you have directly retrieved it and patching it. I will have a look again now after the update. In the meantime feel free to send me a pull request with your changes.

Yeah, I heard that the BD team made lots of efforts dealing with the new changes. Also I've sent a pull request, feel free to check them out :)

**Comment by XHesaX** [comment]
> d0gkiller87

thx man very much

**Comment by Keler69** [comment]
> Here's the working update. Replace the `onStart()` function in `SpotifyListenAlong.plugin.js` with the following snippet:
> 
> ```js
> onStart() {
> 	const getActiveSocketAndDevice = WebpackModules.getByProps( 'getActiveSocketAndDevice' );	
> 	if ( getActiveSocketAndDevice?.getActiveSocketAndDevice ) {
> 		BdApi.Patcher.after(
> 			'', getActiveSocketAndDevice, 'getActiveSocketAndDevice',
> 			( that, args, ret ) => {
> 				if ( ret?.socket ) ret.socket.isPremium = true;
> 				return ret;
> 			}
> 		);
> 	}
> }
> ```
> 
> Best wishes on your exams @ordinall :)

thanks for this, works almost perfectly. the main purpose of the plugin works but it makes bd crash. idk might just be me. i removed it from my plugins folder for now

**Comment by Florian-Mahlberg** [comment]
For me it works, maybe try the new version from #25 ?

**Comment by Keler69** [comment]
> For me it works, maybe try the new version from #25 ?

tried it, working for now. still doing some testing
will update

**Comment by ordinall** [comment]
Rejoice. Fixed in [e3b88c0](https://github.com/ordinall/BetterDiscord-Stuff/commit/e3b88c03a3b30f8c6d56c5335f913909117ac386). Should work now, big thanks to @d0gkiller87, saved me some time.
